File: apps/api/momento_api/models/imports.py
from typing import Optional

from pydantic import BaseModel


class ImportStatusResponse(BaseModel):
    status: str
    total_files: int
    processed_files: int
    successful_imports: int
    failed_imports: int
    started_at: Optional[str]
    completed_at: Optional[str]
    errors: list[str]


class ImportTriggerResponse(BaseModel):
    message: str
    status: str


class RegenerateRequest(BaseModel):
    missing_only: bool = True


class RegenerateResponse(BaseModel):
    message: str
    status: str


class RegenerationStatusResponse(BaseModel):
    status: str
    total_media: int
    processed_media: int
    updated_metadata: int
    generated_thumbnails: int
    updated_tags: int
    started_at: Optional[str]
    completed_at: Optional[str]
    errors: list[str]

File: src/api/src/models/imports.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ImportStatusResponse {
    pub status: String,
    pub total_files: i64,
    pub processed_files: i64,
    pub successful_imports: i64,
    pub failed_imports: i64,
    pub started_at: Option<String>,
    pub completed_at: Option<String>,
    pub errors: Vec<String>,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ImportTriggerResponse {
    pub message: String,
    pub status: String,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RegenerateRequest {
    #[serde(default = "default_missing_only")]
    pub missing_only: bool,
}

fn default_missing_only() -> bool {
    true
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct RegenerateResponse {
    pub message: String,
    pub status: String,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct RegenerationStatusResponse {
    pub status: String,
    pub total_media: i64,
    pub processed_media: i64,
    pub updated_metadata: i64,
    pub generated_thumbnails: i64,
    pub updated_tags: i64,
    pub started_at: Option<String>,
    pub completed_at: Option<String>,
    pub errors: Vec<String>,
}
